File: kappa-recipes/src/test/scala/kappa/recipes/test/DirectTest.scala
package kappa.recipes.test

import kappa._
import kappa.coordinator._
import kappa.Directives._

/**
  * Created by j0rd1 on 10/11/16.
  */
class DirectTest extends Spec {

  val conf = new KappaConf

  override implicit def session: KappaSession = new KappaSession(conf)

  override def domain: Path = "direct_test"

  val start = {
    writeLock() { lock =>
      lock.assertNotLocked("start") {
        executeCommand("") {
          lock.write(s"start-${lock.id}")
        }
      }
    }
  }



}

File: kappa-recipes/src/test/java/kappa/recipes/test/java/DirectTest.java
package kappa.recipes.test.java;

import kappa.KappaSession;
import kappa.Directive;
import kappa.coordinator.Spec;
import static kappa.Directives.*;

/**
 * Created by j0rd1 on 10/11/16.
 */
public class DirectTest extends Spec {

    @Override
    public KappaSession session() {
        return null;
    }

    @Override
    public String domain() {
        return null;
    }

    public final Directive start = writeLock().success(lock ->
        response("")
    );


}
